packages/blocks/google-cloud/src/lib/actions/get-recommendations-cli-action.ts

```diff
@@ -0,0 +1,298 @@
+            },
+          ],
+        };
+      },
+    }),
+    filterByProperty: Property.DynamicProperties({
+      displayName: '',
+      required: true,
+      refreshers: [
+        'auth',
+        'useHostSession',
+        'useHostSession.useHostSessionCheckbox',
+        'filterBySelection',
+      ],
+      props: async ({
+        auth,
+        useHostSession,
+        filterBySelection,
+      }): Promise<{ [key: string]: any }> => {
+        const shouldUseHostCredentials =
+          (useHostSession as { useHostSessionCheckbox?: boolean })
+            ?.useHostSessionCheckbox === true;
+
+        if (!auth && !shouldUseHostCredentials) {
+          return {
+            markdown: Property.MarkDown({
+              value: 'Please authenticate to see the filter options.',
+            }),
+          };
+        }
+
+        if (!filterBySelection) {
+          return {
+            markdown: Property.MarkDown({
+              value:
+                'Please select a filter from the dropdown to see the options.',
+            }),
+          };
+        }
+
+        return await getScopeOptionProperty(
+          auth,
+          shouldUseHostCredentials,
+          filterBySelection,
+        );
+      },
+    }),
+    recommenders: getRecommendersDropdown(),
+    location: Property.ShortText({
+      displayName: 'Location',
+      description: 'Location to list recommendations for.',
+      required: true,
+    }),
+  },
+  async run(context) {
+    const currentCommand = '';
+    try {
+      const { filterByProperty, location, recommenders } = context.propsValue;
+
+      if (!recommenders) {
+        throw new Error('Recommenders are required');
+      }
+
+      let baseCommand = `gcloud recommender recommendations list --format=json ${getFilterByPropertyCommandParam(
+        filterByProperty,
+      )}`;
+
+      if (location) {
+        baseCommand += ` --location=${location}`;
+      }
+
+      const commands = recommenders.map(
+        (recommender) => `${baseCommand} --recommender=${recommender}`,
+      );
+
+      const results = await runCommands(
+        commands,
+        context.auth,
+        context.propsValue.useHostSession?.['useHostSessionCheckbox'],
+      );
+
+      const allRecommendations = results.flatMap((result, i) => {
+        const parsed = tryParseJson(result);
+        return Array.isArray(parsed)
+          ? parsed.map((item) => ({
+              ...item,
+              source: recommenders[i],
+            }))
+          : [];
+      });
+
+      return allRecommendations;
+    } catch (error) {
+      handleCliError({
+        provider: 'Google Cloud',
+        command: currentCommand,
+        error,
+      });
+    }
+  },
+});
+
+async function getScopeOptionProperty(
+  auth: any,
+  useHostSession: any,
+  filterSelection: any,
+) {
+  if (filterSelection === 'folder') {
+    return {
+      folder: Property.ShortText({
+        displayName: 'Folder ID',
```

Comment:
how will the user know this?
is there a way to fetch all folders or something? can we do by folder name?

Reply:
uegh my comment got lost, there is a way to get folders, but it requires the user to either provide orgId or folderId, 

I talked with marco about this and he agreed it makes more sense to just add it as a text, considering we have no way of having the user input the parameters for the request. 

They can find the folder ids in the console